src/excel/Workbook.d.ts

```diff
@@ -93,4 +93,15 @@ export declare interface Workbook {
    * @param name name
    */
   deleteNamedItem(name: string): Promise<void>;
+
+  /**
+   * creates a new session
+   */
+  createSession(): Promise<string>;
```

Comment:
do you also need to [close](https://learn.microsoft.com/en-us/graph/api/workbook-closesession?view=graph-rest-1.0&tabs=http) the session? isn't the risk that too many open sessions cause extra overhead on the sharepoint side?

> Non-persistent session - Changes made by the API are not saved to the source location. Instead, the Excel backend server keeps a temporary copy of the file that reflects the changes made during that particular API session. When the Excel session expires, the changes are lost. This mode is useful for apps that need to do analysis or obtain the results of a calculation or a chart image, but not affect the document state.

also, it looks like you don't set `persistChanges=true`, so I'd assume that changes to such sessions are not written back to excel.

Reply:
This will create a persistent session as by default, the session returned has persistChanges = true in the response.
According to: https://learn.microsoft.com/en-us/graph/api/resources/excel?view=graph-rest-1.0#sessions-and-persistence, the persistent session automatically expires after 5 minutes of activity.
So, we don't need to close the session.

Reply:
We'll need a refresh session call, which I'm planning to contribute as the next step.

Reply:
> the persistent session automatically expires after 5 minutes of inactivity.
> So, we don't need to close the session.

but repeated updates on the same workbook will generate more and more sessions. I think it would be  good behavior to close the session when you don't need it anymore.

Reply:
> but repeated updates on the same workbook will generate more and more sessions

As the same session will be refreshed periodically from the client, we will not create more sessions (during authoring).

> I think it would be good behavior to close the session when you don't need it anymore.

May be we can do this if the user closes the UE window (separate API call to service to close the session) - but wouldn't that be overkill?

Reply:
> As the same session will be refreshed periodically from the client, we will not create more sessions (during authoring).

ah, ok. but for completeness, I would add `closeSession` to the API.

Reply:
I've added both closeSession and refreshSession APIs for completeness. We'd anyways need refreshSession.